instrumentation/opentelemetry-instrumentation-falcon/tests/test_falcon.py

```diff
@@ -101,17 +99,20 @@ def _test_method(self, method):
                 SpanAttributes.NET_HOST_PORT: 80,
                 SpanAttributes.HTTP_HOST: "falconframework.org",
                 SpanAttributes.HTTP_TARGET: "/",
-                SpanAttributes.NET_PEER_IP: "127.0.0.1",
                 SpanAttributes.NET_PEER_PORT: "65133",
                 SpanAttributes.HTTP_FLAVOR: "1.1",
                 "falcon.resource": "HelloWorldResource",
                 SpanAttributes.HTTP_STATUS_CODE: 201,
             },
         )
+        if SpanAttributes.NET_PEER_IP in span.attributes:
+            self.assertEqual(
+                span.attributes[SpanAttributes.NET_PEER_IP], "127.0.0.1"
```

Comment:
Curious, if `remote_addr="127.0.0.1"` was removed from the call to `simulate_request` then why should it appear here? Is this a default value? If so, it would be good to test also a non-default value.

Reply:
@ocelotl,

In falcon 1.4.1 (the new version), the remote_addr param doesn't exist. This to me wasn't a huge deal as in the tests, it was being tested with the default value anyways, '127.0.0.1'.

However in falcon 3.x, there was an update to the code that was made which doesn't set the remote_addr if the default value is used. (See my comment above) or

https://github.com/falconry/falcon/blob/2.0.0/falcon/testing/helpers.py#L176
vs
https://github.com/falconry/falcon/blob/3.0.1/falcon/testing/helpers.py#L1168-L1172


 This is different to how it was done in falcon 2.x where the remote_addr was always set, just with the default value.
 
 The if statement in the test here was a compromise to show that there is something strange going on between the different versions.
 
 If you have any thoughts on this, please let me know. I am open to suggestions.

Reply:
Ok, that's fine, can you please add this explanation to the test as a comment? :v:

Reply:
Added. Thank you